packages/storefront-api-client/src/tests/storefront-api-client.test.ts

```diff
@@ -0,0 +1,471 @@
+                ", "
+              )}`
+            )
+          );
+        });
+
+        it("console warns when a unsupported api version is provided", () => {
+          const consoleWarnSpy = jest
+            .spyOn(window.console, "warn")
+            .mockImplementation(jest.fn());
+
+          createStorefrontAPIClient({
+            ...config,
+            apiVersion: "2022-07",
+          });
+
+          expect(consoleWarnSpy).toHaveBeenCalledWith(
+            `Storefront API Client: the provided \`apiVersion\` (\`2022-07\`) is deprecated or not supported. Current supported API versions: ${mockApiVersions.join(
+              ", "
+            )}`
+          );
+        });
+
+        it("throws an error when neither public and private access tokens are provided", () => {
+          expect(() =>
+            // @ts-ignore
+            createStorefrontAPIClient({
+              ...config,
+              publicAccessToken: undefined,
+            })
+          ).toThrow(
+            new Error(
+              `Storefront API Client: a public or private access token must be provided`
+            )
+          );
+        });
+
+        it("throws an error when a private access token is provided in a browser environment (window is defined)", () => {
+          expect(() =>
+            // @ts-ignore
+            createStorefrontAPIClient({
+              ...config,
+              publicAccessToken: undefined,
+              privateAccessToken: "private-access-token",
+            })
+          ).toThrow(
+            new Error(
+              "Storefront API Client: private access tokens and headers should only be used in a server-to-server implementation. Use the API public access token in nonserver environments."
+            )
+          );
+        });
+
+        it("throws an error when both public and private access tokens are provided in a server environment (window is undefined)", () => {
+          const windowSpy = jest
+            .spyOn(window, "window", "get")
+            .mockImplementation(() => undefined as any);
+
+          expect(() =>
+            // @ts-ignore
+            createStorefrontAPIClient({
+              ...config,
+              privateAccessToken: "private-token",
+            })
+          ).toThrow(
+            new Error(
+              `Storefront API Client: only provide either a public or private access token`
+            )
+          );
+
+          windowSpy.mockRestore();
+        });
+      });
+    });
+
+    describe("client config", () => {
+      it("returns a config object that includes the provided store domain", () => {
+        const client = createStorefrontAPIClient(config);
+        expect(client.config.storeDomain).toBe(config.storeDomain);
+      });
+
+      it("returns a config object that includes the provided public access token, a null private access token and a header property for the public token", () => {
```

Comment:
```suggestion
      it("returns a config object that includes the provided public access token and a null private access token", () => {
```

Think this might be a typo after refactoring - header checks seem further down.